### PR TITLE
Array, String and Encoding wrappers extended with new methods

### PR DESCRIPTION
**Pull request by niku-live**
Array wrapper extended with methods to create and manipulate Byte, Char, Int32 and String arrays (all 4 types of array is frequently used, especially when working with streams and text).

String wrapper extended with new methods to manipulate text. I know, that AL has its own Text datatype with similar methods, but these few methods I added are frequently used and it is very strange to always convert to/from DotNetString wrapper just to call native methods.

Encoding wrapper extended with methods to get string directly from byte array, this is very useful when implementing binary data file formats import functionality.

Test codeunits iliustrate how to use all of those new methods.

### CODE REVIEW

**Comment by GiorgosPa** [comment]
Thanks for the suggestion! We’re investigating whether we can use it and hope to provide an answer soon.

File: Test/COD50300.TXT [hunk 1]
```diff
@@ -0,0 +1,153 @@
+OBJECT Codeunit 50300 Test_DotNet_Encoding
```
Comment (GiorgosPa) [inline Test/COD50300.TXT]: The tests should be moved in Codeunit 146026

File: Test/COD50302.TXT [hunk 2]
```diff
@@ -0,0 +1,161 @@
+OBJECT Codeunit 50302 Test_DotNet_String
```
Comment (GiorgosPa) [inline Test/COD50302.TXT]: Please re-id to 146007

File: Test/COD50301.TXT [hunk 3]
```diff
@@ -0,0 +1,143 @@
+OBJECT Codeunit 50301 Test_DotNet_Array
```
Comment (GiorgosPa) [inline Test/COD50301.TXT]: Please re-id to 146000

File: BaseApp/COD3000.TXT [hunk 4]
```diff
@@ -17,18 +17,100 @@ OBJECT Codeunit 3000 DotNet_Array
...
+    BEGIN
+      EXIT(DotNetArray.GetValue(Index));
+    END;
+
+    [External]
+    PROCEDURE GetValueAsInteger@106(Index@1000 : Integer) : Integer;
+    BEGIN
+      EXIT(DotNetArray.GetValue(Index));
+    END;
+
```
Comment (GiorgosPa) [inline BaseApp/COD3000.TXT]: Add A GetValueAsByte for completenes

File: Test/COD50300.TXT (hunk 1 above)
Comment (niku-live) [reply Test/COD50300.TXT]: Fixed with latest commits

File: BaseApp/COD3000.TXT (hunk 4 above)
Comment (niku-live) [reply BaseApp/COD3000.TXT]: Fixed with latest commits

File: Test/COD50301.TXT (hunk 3 above)
Comment (niku-live) [reply Test/COD50301.TXT]: Fixed with latest commits

File: Test/COD50302.TXT (hunk 2 above)
Comment (niku-live) [reply Test/COD50302.TXT]: Fixed with latest commits

**Comment by niku-live** [comment]
Hi,

With my latest commits I moved Test codeunits to correct IDs also implemented GetValueAsByte for completeness. About case sensitive problems - I didn't even noticed that in my repository files were with wrong case. I tried fixing this by removing files and adding them again. At least on my local repository now everything look clear and ok, but write me if something is wrong.

**Pull request merged**